ios/Classes/TiMapView.m

```diff
@@ -895,21 +895,47 @@ - (TiMapAnnotationProxy*)proxyForAnnotation:(MKAnnotationView*)pinview
 	return nil;
 }
 
-- (void)mapView:(MKMapView *)mapView didSelectAnnotationView:(MKAnnotationView *)view{
-	if ([view conformsToProtocol:@protocol(TiMapAnnotation)])
-	{
+- (void)mapView:(MKMapView *)mapView didSelectAnnotationView:(MKAnnotationView *)view
+{
+    if ([view conformsToProtocol:@protocol(TiMapAnnotation)])
+    {
+        BOOL isSelected = [view isSelected];
+        MKAnnotationView<TiMapAnnotation> *ann = (MKAnnotationView<TiMapAnnotation> *)view;
         
-		BOOL isSelected = [view isSelected];
-		MKAnnotationView<TiMapAnnotation> *ann = (MKAnnotationView<TiMapAnnotation> *)view;
-		[self fireClickEvent:view source:isSelected?@"pin":[ann lastHitName]];
+        selectedAnnotation = ann;
+        
+        //if canShowCallout == true we will try to find calloutView to hadleTap on callout
+        if ([ann canShowCallout]) {
+            //run after delay because we have to wait for animation finish
+            [self performSelector:@selector(findCalloutView:) withObject:ann afterDelay:0.2];
```

Comment:
do you meant something like this

            dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(0.2f * NSEC_PER_SEC)), dispatch_get_main_queue(), ^(void){
                [self findCalloutView: ann];
            });